src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -147,21 +141,26 @@ def _generic_method_handler(
                     PingRequest,
                 ),
             ):
+                # Verify hmac value
                 hmac_value = base64.urlsafe_b64decode(
                     _get_value_from_tuples(
                         _AUTH_TOKEN_HEADER, context.invocation_metadata()
                     )
                 )
-                client_public_key = bytes_to_public_key(client_public_key_bytes)
-                shared_secret = generate_shared_key(
-                    self.server_private_key,
-                    client_public_key,
-                )
-                verify = verify_hmac(
-                    shared_secret, request.SerializeToString(True), hmac_value
-                )
-                if not verify:
+                public_key = bytes_to_public_key(client_public_key_bytes)
+
+                if not self._verify_hmac(public_key, request, hmac_value):
                     context.abort(grpc.StatusCode.UNAUTHENTICATED, "Access denied")
+
+                # Verify node_id
+                try:
+                    node_id = self.state.get_node_id(client_public_key_bytes)
+                except KeyError:
+                    node_id = -1
```

Comment:
Instead of a magic number, let's make `node_id` an `Optional[int]` (only in the scope of this function, of course) and set it to `None` here.

Reply:
Removed magic number.